Lib/tty.py

```diff
@@ -15,22 +24,81 @@
 OSPEED = 5
 CC = 6
 
-def setraw(fd, when=TCSAFLUSH):
-    """Put terminal into a raw mode."""
-    mode = tcgetattr(fd)
-    mode[IFLAG] = mode[IFLAG] & ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
-    mode[OFLAG] = mode[OFLAG] & ~(OPOST)
-    mode[CFLAG] = mode[CFLAG] & ~(CSIZE | PARENB)
-    mode[CFLAG] = mode[CFLAG] | CS8
-    mode[LFLAG] = mode[LFLAG] & ~(ECHO | ICANON | IEXTEN | ISIG)
+try:
+    CHECK_WINSZ = TIOCGWINSZ
+    CHECK_WINSZ = TIOCSWINSZ
+    HAVE_WINSZ = True
+
+    def getwinsz(fd):
+        """Gets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        s = pack("HHHH", 0, 0, 0, 0)
+        return ioctl(fd, TIOCGWINSZ, s)
+
+    def setwinsz(fd, winsz):
+        """Sets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        ioctl(fd, TIOCSWINSZ, winsz)
+except NameError:
+    HAVE_WINSZ = False
+    getwinsz = None
```

Comment:
I think better to make a stub that raises `NotImplementedError` is better than setting the function name to `None`.
In std lib we never use this pattern.

Sometimes we don't declare unsupported function at all, but in this case we usually mark in docs: "Availability: Windows" or "Availability: Linux kernel >= 4.5 or glibc >= 2.27".

I think here is better to support stubs that always fail.

Reply:
I am working on this.